tests/nested.rs

```diff
@@ -91,3 +91,34 @@ fn nested_middleware() {
     block_on(res.into_body().read_to_end(&mut buf)).unwrap();
     assert_eq!(&*buf, &*b"/bar");
 }
+
+#[test]
+fn nested_with_different_state() {
+    let mut outer = tide::new();
+    let mut inner = tide::with_state(42);
+    inner.at("/").get(|req: tide::Request<i32>| async move {
+        let num = req.state();
+        format!("the number is {}", num)
+    });
+    outer.at("/").get(|_| async move { "Hello, world!" });
+    outer
+        .at("/foo")
+        .strip_prefix()
+        .get(inner.into_http_service());
```

Comment:
I've added a simpler method to nest services. This can be simplified to `.at().nest(inner)` now; it would be good to rebase and update!